File: src/dataset/python/beer-song_exercism-fruit-in/claude/ClaudeSonnet4_beer_song_v2.py
def recite(start, take=1):
    if take == 0:
        return []
    
    result = []
    for i in range(take):
        n = start - i
        bottle_text = bottle(n)
        bottle_lower = bottle_text.lower()
        next_bottle = bottle((n + 99) % 100).lower()
        
        result.extend([
            f"{bottle_text} of beer on the wall, {bottle_lower} of beer.",
            f"{take_down(n)}, {next_bottle} of beer on the wall.",
            ""
        ])
    
    return result[:-1]


def verse(n):
    bottle_text = bottle(n)
    bottle_lower = bottle_text.lower()
    next_bottle = bottle((n + 99) % 100).lower()
    
    return [
        f"{bottle_text} of beer on the wall, {bottle_lower} of beer.",
        f"{take_down(n)}, {next_bottle} of beer on the wall.",
        "",
    ]


def take_down(n):
    if n == 0:
        return "Go to the store and buy some more"
    elif n == 1:
        return "Take it down and pass it around"
    else:
        return "Take one down and pass it around"


def bottle(n):
    if n == 0:
        return "No more bottles"
    elif n == 1:
        return "1 bottle"
    else:
        return f"{n} bottles"

File: src/dataset/c/beer-song_exercism-IrvinRamiA/claude/ClaudeSonnet4_beer_song_v2.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "beer_song.h"

void recite(uint8_t start_bottles, uint8_t take_down, char **song)
{
    uint16_t song_index = 0;
    uint8_t current_bottles = start_bottles;
    
    for (uint8_t verse = 0; verse < take_down; verse++)
    {
        if (current_bottles == 2)
        {
            sprintf(song[song_index++], "2 bottles of beer on the wall, 2 bottles of beer.");
            sprintf(song[song_index++], "Take one down and pass it around, 1 bottle of beer on the wall.");
        }
        else if (current_bottles == 1)
        {
            sprintf(song[song_index++], "1 bottle of beer on the wall, 1 bottle of beer.");
            sprintf(song[song_index++], "Take it down and pass it around, no more bottles of beer on the wall.");
        }
        else if (current_bottles == 0)
        {
            sprintf(song[song_index++], "No more bottles of beer on the wall, no more bottles of beer.");
            sprintf(song[song_index++], "Go to the store and buy some more, 99 bottles of beer on the wall.");
        }
        else
        {
            sprintf(song[song_index++], "%d bottles of beer on the wall, %d bottles of beer.", current_bottles, current_bottles);
            sprintf(song[song_index++], "Take one down and pass it around, %d bottles of beer on the wall.", current_bottles - 1);
        }
        
        if (verse < take_down - 1)
        {
            song[song_index++][0] = '\0';
        }
        
        current_bottles--;
    }
}